File: django_api/jobs/models/job_details.py
from django.db import models

class JobDetails(models.Model):
    job = models.OneToOneField(
        "jobs.Job",
        on_delete=models.CASCADE,
        related_name="details"
    )
    description = models.TextField()
    requirements = models.JSONField(default=dict)
    responsibilities = models.JSONField(default=dict)
    external_apply = models.BooleanField(default=False)
    apply = models.URLField(blank=True, null=True)
    experience_required = models.CharField(max_length=100, blank=True, null=True)
    
    def __str__(self):
        return f"Details for {self.job.position} at {self.job.company}"

File: MERN/server/models/JobDetails.ts
import mongoose, { Document, Schema, Types } from 'mongoose';

export interface IJobDetails extends Document {
  job: Types.ObjectId;
  description: string;
  requirements: {
    content: string;
    items: string[];
  };
  responsibilities: {
    content: string;
    items: string[];
  };
  externalApply: boolean;
  apply?: string;
  experienceRequired?: string;
}

const JobDetailsSchema = new Schema<IJobDetails>(
  {
    job: {
      type: Schema.Types.ObjectId,
      ref: 'Job',
      required: true,
      unique: true,
    },
    description: {
      type: String,
      required: [true, 'Description is required'],
    },
    requirements: {
      content: {
        type: String,
        default: '',
      },
      items: {
        type: [String],
        default: [],
      },
    },
    responsibilities: {
      content: {
        type: String,
        default: '',
      },
      items: {
        type: [String],
        default: [],
      },
    },
    externalApply: {
      type: Boolean,
      default: false,
    },
    apply: {
      type: String,
    },
    experienceRequired: {
      type: String,
      maxlength: 100,
    },
  },
  {
    timestamps: false,
    collection: 'job_details',
  }
);

export const JobDetails = mongoose.model<IJobDetails>('JobDetails', JobDetailsSchema);
